c_emulator/riscv_callbacks_if.cpp

```diff
@@ -0,0 +1,121 @@
+#include "riscv_callbacks_if.h"
+
+#include "sail_riscv_model.h"
+
+// Default implementations that do nothing.
+
+// Callback invoked before each step
+void callbacks_if::pre_step_callback([[maybe_unused]] hart::Model &model,
+                                     [[maybe_unused]] bool is_waiting)
+{
+}
```

Comment:
```suggestion
void callbacks_if::pre_step_callback([hart::Model &, bool) {}
```
Could drop the name which makes it a lot shorter? Now that this is in a separate file the header has the name and we don't necessarily need it here?